android/src/main/java/com/reactnativefileutils/FileUtilsModule.java

```diff
@@ -25,141 +25,170 @@
 
 @ReactModule(name = FileUtilsModule.NAME)
 public class FileUtilsModule extends ReactContextBaseJavaModule {
-    public static final String NAME = "FileUtils";
+  public static final String NAME = "FileUtils";
+
+  public FileUtilsModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  @NonNull
+  public String getName() {
+    return NAME;
+  }
+
+  /**
+   * Gets the duration of a video in seconds.
+   *
+   * @param uri - The video file path to get the duration of.
+   * @returns The duration in seconds of the video file.
+   */
+  @ReactMethod
+  public void getDuration(String uri, Promise promise) {
+    try {
+      Uri fileUri = Uri.parse(uri);
+      FileDescriptor fileDescriptor = getReactApplicationContext().getContentResolver().openAssetFileDescriptor(fileUri, "r").getFileDescriptor();
+      MediaMetadataRetriever retriever = new MediaMetadataRetriever();
 
-    public FileUtilsModule(ReactApplicationContext reactContext) {
-        super(reactContext);
-    }
+      retriever.setDataSource(fileDescriptor);
+      String time = retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+      float durationInSeconds = Float.parseFloat(time) / 1000;
+      retriever.release();
 
-    @Override
-    @NonNull
-    public String getName() {
-        return NAME;
+      promise.resolve(String.valueOf(durationInSeconds));
+    } catch (Exception e) {
+      promise.reject("Error getting duration of video", e);
     }
+  }
+
+  /**
+   * Gets the pixel dimensions, height and width (x,y), of the video or image file based on the
+   * file path passed in.
+   *
+   * @param uri       - The video or image file path to get the dimensions of.
+   * @param mediaType - Either 'video' or 'image' so the method knows how to process the media file.
+   * @returns The height and width (x,y), of the video or image in pixels.
+   */
+  @ReactMethod
+  public void getDimensions(String uri, String mediaType, Promise promise) {
+    Uri fileUri = Uri.parse(uri);
+    try {
+
+      // Handle getting image dimensions
+      if (mediaType.equalsIgnoreCase("image")) {
```

Comment:
For getting Dimensions of an Image, should we not handle this on the JS layer with `Image.getSize` which is part of the core RN API? So something like:

```
function getDimensions() {
     if(video) {
           NativeModule.getDimentions()
     } else { 
          Image.getSize(url, ...);     
     }
}
```

Reply:
Oh nice, yes, less for us to maintain. I switched this core util out to handle image dimensions.